pkgs/os-specific/linux/ippcrypto/default.nix

```diff
@@ -0,0 +1,90 @@
+{ lib,
+  stdenvNoCC,
+  fetchurl,
+  fetchFromGitHub,
+  autoconf,
+  automake,
+  binutils,
+  bison,
+  cmake,
+  file,
+  flex,
+  gcc,
+  git,
+  gnumake,
+  gnum4,
+  libtool,
+  nasm,
+  ocaml,
+  ocamlPackages,
+  openssl,
+  perl,
+  python3,
+  texinfo
+}:
+
+stdenvNoCC.mkDerivation {
+  pname = "ippcrypto";
+  version = "ippcp_2020u3";
+  src = fetchFromGitHub {
+    owner = "sbellem";
+    repo = "linux-sgx";
+    rev = "e6036d6edb4371f2acc64c50b7cb51e9dfa439a4";
+    sha256 = "0znallianv3lp3y62cfdgp8gacpw516qg8cjxhz8bj5lv5qghchk";
+    fetchSubmodules = true;
+  };
+  buildInputs = [
+    binutils
+    autoconf
+    automake
+    libtool
+    ocaml
+    ocamlPackages.ocamlbuild
+    file
+    cmake
+    gnum4
+    openssl
+    gcc
+    gnumake
+    texinfo
+    bison
+    flex
+    perl
+    python3
+    git
+    nasm
+  ];
+  dontConfigure = true;
+  # sgx expects binutils to be under /usr/local/bin by default
+  preBuild = ''
+    export BINUTILS_DIR=$binutils/bin
+    '';
+  buildPhase = ''
+    runHook preBuild
```

Comment:
I think hooks run by default, was that needed?

Reply:
If you overwrite the phase they are not run by default.